File: tests/machine/x/dart/string_prefix_slice.dart
var prefix = 'fore';

var s1 = 'forest';

var s2 = 'desert';

void main() {
  print(
    ((s1 is String)
            ? s1.substring(0, prefix.length)
            : (s1 as List).sublist(0, prefix.length)) ==
        prefix,
  );
  print(
    ((s2 is String)
            ? s2.substring(0, prefix.length)
            : (s2 as List).sublist(0, prefix.length)) ==
        prefix,
  );
}

File: tests/machine/x/java/string_prefix_slice.java
import java.util.*;
public class Main {
	static String prefix = "fore";
	static String s1 = "forest";
	static String s2 = "desert";
	static <K,V> LinkedHashMap<K,V> mapOf(Object... kv) {
		LinkedHashMap<K,V> m = new LinkedHashMap<>();
		for (int i = 0; i < kv.length; i += 2) m.put((K)kv[i], (V)kv[i+1]);
		return m;
	}
	public static void main(String[] args) {
	System.out.println(Objects.equals(s1.substring(0, prefix.length()), prefix));
	System.out.println(Objects.equals(s2.substring(0, prefix.length()), prefix));
	}
}
